protoc-gen-pbandk/lib/src/commonMain/kotlin/pbandk/gen/Namer.kt

```diff
@@ -52,11 +52,14 @@ public interface Namer {
 
         override fun newEnumValueTypeName(enumTypeName: String, preferred: String, nameSet: Collection<String>): String {
             val typePrefix = splitWordsToSnakeCase(enumTypeName) + '_'
-            var name = splitWordsToSnakeCase(preferred).substringAfter(typePrefix)
-            name = name.uppercase()
+            var name = splitWordsToSnakeCase(preferred).let { snakeCase ->
+                snakeCase.substringAfter(typePrefix).takeUnless {
+                    "^\\d|\\W".toRegex().matches(it)
```

Comment:
wasn't sure about adding the `\W` for non-word characters. i wasn't able to test it since if I tried to add a weird character in the protobuf the project wouldn't compile, but figured I'd leave it for good measure just in case.